backend/database.js

```diff
@@ -52,57 +52,91 @@ export default ((sbp('sbp/selectors/register', {
     }
     // Number of entries pushed.
     let counter = 0
-    let { hash: currentHash, ...serverMeta } = (
-      await sbp('chelonia/db/getEntryMeta', contractID, height)
-    ) || {}
+    let currentHeight = height
+    let currentHash, serverMeta
     let prefix = ''
-    let ended = false
-    // NOTE: if this ever stops working you can also try Readable.from():
-    // https://nodejs.org/api/stream.html#stream_stream_readable_from_iterable_options
-    const stream = new Readable({
-      construct (cb) {
-        this.push('[')
-        cb()
-      },
-      read (): void {
-        if (ended) {
-          this.destroy()
-          return
+    const nextKeyOp = (() => {
```

Comment:
Above this could you please add a comment explaining the function's behavior as well as all possible return values and their corresponding meaning?